src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -5,68 +5,122 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
-using NuGet.Common;
-using NuGet.Protocol;
+using Microsoft.DotNet.DotNetSdkResolver;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
 
 namespace Microsoft.DotNet.TemplateLocator
 {
     public sealed class TemplateLocator
     {
-        private DirectoryInfo _dotnetSdkTemplatesLocation;
+        private readonly Lazy<NETCoreSdkResolver> _netCoreSdkResolver;
 
         public TemplateLocator()
+            : this(Environment.GetEnvironmentVariable, VSSettings.Ambient)
         {
-            string mockTemplateLocation = Environment.GetEnvironmentVariable("MOCKDOTNETSDKTEMPLATESLOCATION");
-            if (!string.IsNullOrWhiteSpace(mockTemplateLocation))
-            {
-                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockTemplateLocation);
-            }
         }
 
-        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        /// <summary>
+        /// Test constructor
+        /// </summary>
+        public TemplateLocator(Func<string, string> getEnvironmentVariable, VSSettings vsSettings)
+        {
+            _netCoreSdkResolver =
+                new Lazy<NETCoreSdkResolver>(() => new NETCoreSdkResolver(getEnvironmentVariable, vsSettings));
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(
+            string sdkVersion,
+            string dotnetRootPath)
         {
-            if (_dotnetSdkTemplatesLocation == null)
+            if (string.IsNullOrWhiteSpace(sdkVersion))
             {
-                return Array.Empty<IOptionalSdkTemplatePackageInfo>();
+                throw new ArgumentException($"'{nameof(sdkVersion)}' cannot be null or whitespace", nameof(sdkVersion));
+            }
+
+            if (string.IsNullOrWhiteSpace(dotnetRootPath))
+            {
+                throw new ArgumentException($"'{nameof(dotnetRootPath)}' cannot be null or whitespace", nameof(dotnetRootPath));
+            }
+
+            if (!Version.TryParse(sdkVersion.Split('-')[0], out var sdkVersionParsed))
+            {
+                throw new ArgumentException($"'{nameof(sdkVersion)}' should be a version but get {sdkVersion}");
```

Comment:
Missing comma before "but". Should probably be "got" instead of "get"

Reply:
fixed